paimon-core/src/main/java/org/apache/paimon/append/AppendOnlyCompactionTask.java

```diff
@@ -39,13 +40,22 @@ public class AppendOnlyCompactionTask {
     private final List<DataFileMeta> compactBefore;
     private final List<DataFileMeta> compactAfter;
 
+    private final Identifier tableIdentifier;
+
     public AppendOnlyCompactionTask(BinaryRow partition, List<DataFileMeta> files) {
+        this(partition, files, Identifier.EMPTY);
+    }
+
+    public AppendOnlyCompactionTask(
+            BinaryRow partition, List<DataFileMeta> files, Identifier identifier) {
```

Comment:
I think a new class `MultiTableAppendOnlyCompactionTask` which extends `AppendOnlyCompactionTask` seems better, in this way, we don't need Identifier.EMPTY which may cause hidden bug.

Reply:
We can just put string in `MultiTableAppendOnlyCompactionTask`, then we can remove `IdentifierSerializer`.

Reply:
done ,this change has been finished at #3174